opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorForJavaTest.java

```diff
@@ -92,4 +114,57 @@ public void testGetValuesStrShouldGetCorrectlyFormattedString() throws Exception
         assertEquals("val1", valStr);
     }
+
+    @Test
+    public void testGetClientAttributesShouldReturnRequiredValues() throws Exception {
+        ClientProcessorForJava clientProcessor = new ClientProcessorForJava(context);
+        Map<String, Object> attributes = new HashMap<>();
+        attributes.put("national_id", "3434-34");
+        attributes.put("drivers-license", "DL-324");
+        Client client = new Client("123-23");
+        client.setAttributes(attributes);
+        Map<String, Object> result = Whitebox.invokeMethod(clientProcessor, "getClientAttributes", client);
+        assertEquals(attributes, result);
+    }
+
+    @Test
+    public void testGetGenderShouldReturnCorrectValue() throws Exception {
+        ClientProcessorForJava clientProcessor = new ClientProcessorForJava(context);
+        Client client = new Client("123-23");
+        client.setGender("Female");
+        Map<String, String> resultMap = Whitebox.invokeMethod(clientProcessor, "getGender", client);
+        Map<String, String> expectedMap = new HashMap<>();
+        expectedMap.put("gender", "Female");
+        assertEquals(expectedMap, resultMap);
+    }
+
+    @Test
+    public void testUpdateIdenitifierShouldRemoveHyphenFromOpenmrsId() throws Exception {
+        ClientProcessorForJava clientProcessor = new ClientProcessorForJava(context);
+        ContentValues contentValues = new ContentValues();
+        contentValues.put("zeir_id", "23-23");
+        Whitebox.setInternalState(clientProcessor, "openmrsGenIds", new String[]{"zeir_id"});
+        Whitebox.invokeMethod(clientProcessor, "updateIdenitifier", contentValues);
+        assertEquals("2323", contentValues.get("zeir_id"));
+    }
+
+    @Test
+    public void testCloseCaseShouldReturnFalseIfCloseCaseIsEmpty() {
+        ClientProcessorForJava clientProcessor = new ClientProcessorForJava(context);
+        assertFalse(clientProcessor.closeCase(new Client("1233-2"), new ArrayList<>()));
+    }
+
+    @Test
+    public void testCloseCaseShouldPassCorrectValuesToCloseCase() {
+        ClientProcessorForJava clientProcessor = new ClientProcessorForJava(context);
+        CommonRepository commonRepository = Mockito.mock(CommonRepository.class);
+        ReflectionHelpers.setStaticField(CoreLibrary.class, "instance", coreLibrary);
+        PowerMockito.when(coreLibrary.context()).thenReturn(opensrpContext);
+        PowerMockito.when(opensrpContext.commonrepository("child")).thenReturn(commonRepository);
+        assertTrue(clientProcessor.closeCase(new Client("1233-2"), Arrays.asList("child")));
+        Mockito.verify(commonRepository).closeCase(closeCaseArgumentCaptor.capture(), closeCaseArgumentCaptor.capture());
+        assertEquals("1233-2", closeCaseArgumentCaptor.getAllValues().get(0));
+        assertEquals("child", closeCaseArgumentCaptor.getAllValues().get(1));
+        ReflectionHelpers.setStaticField(CoreLibrary.class, "instance", null);
```

Comment:
why is this required, unit tests should normally end with asserts or expected failures

Reply:
let add a teardown for this